[choe] replace scraperhelper.NewScraperWithComponentType to scraperhelper.NewScraper as deprecated

mrasu opened the pull request:
To change an argument of `scraperhelper.NewScraper`, `scraperhelper.NewScraperWithComponentType` is introduced in previous version and deprecated in newer version at https://github.com/open-telemetry/opentelemetry-collector/pull/11159.

This PR update dependencies by `multimod sync` and replace `scraperhelper.NewScraperWithComponentType` to `scraperhelper.NewScraper`

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/9473

crobert-1 commented on cmd/otelcontribcol/go.mod:
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 go 1.22.0
 
-toolchain go1.22.7
+toolchain go1.23.1
```
We shouldn't update the toolchain version here. (Same comment for other toolchain updates)

crobert-1 commented on cmd/githubgen/go.mod:
```diff
@@ -4,11 +4,13 @@ go 1.22.0
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+require github.com/rogpeppe/go-internal v1.12.0 // indirect
```
Can you share why this dependency is required? I wouldn't have expected a new dependency to be required when the only change is updating existing dependencies, but one of them must require it now?

mrasu committed:
update opentelemetry-collector dependencies

mrasu committed:
replace scraperhelper.NewScraperWithComponentType to scraperhelper.NewScraper as deprecated

mrasu commented:
@crobert-1 
Sorry. I ran `make update-otel` but encountered errors saying `go.opentelemetry.io/collector/internal/localhostgate` was not found during gotidy.
So, I fixed this by running `go mod edit -require go.opentelemetry.io/collector/config/internal@v0.109.1-0.20240916093651-6029f3176705` and then executed the commands after gotidy in `update-otel` manually.
However, it made huge diffs including unrelated changes as you mentioned.

So, I recreated the changes by running `multimod sync -s=true -o ../opentelemetry-collector -m beta,stable --commit-hash 6029f31767` and fixed the errors manually.

mrasu committed:
specify commit to go.opentelemetry.io/collector/internal/globalgates

mrasu committed:
replace scraperhelper.NewScraperWithComponentType to scraperhelper.NewScraper as deprecated

mx-psi commented:
Overlaps with #35215, I think @codeboten wants to merge #35215 first

codeboten commented:
sorry @mrasu! i hadnt seen this PR before i started working on my own update core PR to include the changes i made to the processor helper... i think it's almost ready to merge

mrasu commented:
@codeboten No worries! close this
